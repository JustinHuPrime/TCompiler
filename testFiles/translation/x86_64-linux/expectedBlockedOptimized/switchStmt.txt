testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    J2B(LOCAL(3), LOCAL(18), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
  ),
  BLOCK(9,
    JUMP(LOCAL(13)),
  ),
  BLOCK(13,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(10,
    JUMP(LOCAL(14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(11,
    JUMP(LOCAL(15)),
  ),
  BLOCK(15,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(18,
    J2A(LOCAL(3), LOCAL(19), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
  ),
  BLOCK(19,
    NOP(),
    NOP(),
    NOP(),
    NOP(),
    JUMPTABLE(TEMP(temp22, 8, 8, GP), LOCAL(17)),
  ),
  BLOCK(4,
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
  ),
)
RODATA(LOCAL(17), 8,
  LABEL(LOCAL(9)),
  LABEL(LOCAL(11)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
)
