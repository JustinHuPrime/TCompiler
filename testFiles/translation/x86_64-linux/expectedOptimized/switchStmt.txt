testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    JB(LOCAL(3), LOCAL(18), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
  ),
  BLOCK(18,
    JA(LOCAL(3), LOCAL(19), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
  ),
  BLOCK(19,
    SUB(TEMP(temp20, 1, 1, GP), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    ZX(TEMP(temp23, 8, 8, GP), TEMP(temp20, 1, 1, GP)),
    UMUL(TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP), CONSTANT(8, LONG(8))),
    ADD(TEMP(temp22, 8, 8, GP), TEMP(temp21, 8, 8, GP), LOCAL(17)),
    JUMP(TEMP(temp22, 8, 8, GP)),
  ),
)
RODATA(.L17, 8,
  LABEL(.L9),
  LABEL(.L11),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
)
