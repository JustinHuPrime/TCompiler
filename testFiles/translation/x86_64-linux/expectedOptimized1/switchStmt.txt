testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    J2B(LOCAL(3), LOCAL(18), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
  ),
  BLOCK(9,
    JUMP(LOCAL(13)),
  ),
  BLOCK(13,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(10,
    JUMP(LOCAL(14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(11,
    JUMP(LOCAL(15)),
  ),
  BLOCK(15,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    JUMP(LOCAL(4)),
  ),
  BLOCK(18,
    J2A(LOCAL(3), LOCAL(19), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
  ),
  BLOCK(19,
    SUB(TEMP(temp20, 1, 1, GP), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    ZX(TEMP(temp23, 8, 8, GP), TEMP(temp20, 1, 1, GP)),
    UMUL(TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP), CONSTANT(8, LONG(8))),
    ADD(TEMP(temp22, 8, 8, GP), TEMP(temp21, 8, 8, GP), LOCAL(17)),
    JUMP(TEMP(temp22, 8, 8, GP)),
  ),
  BLOCK(4,
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
  ),
)
RODATA(LOCAL(17), 8,
  LABEL(.L9),
  LABEL(.L11),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
)
