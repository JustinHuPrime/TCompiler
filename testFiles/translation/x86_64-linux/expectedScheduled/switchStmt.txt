testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    LABEL(CONSTANT(8, LOCAL(5))),
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    LABEL(CONSTANT(8, LOCAL(3))),
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    J1B(CONSTANT(8, LOCAL(3)), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    LABEL(CONSTANT(8, LOCAL(18))),
    J1A(CONSTANT(8, LOCAL(3)), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
    LABEL(CONSTANT(8, LOCAL(19))),
    NOP(),
    NOP(),
    NOP(),
    NOP(),
    JUMPTABLE(TEMP(temp22, 8, 8, GP), CONSTANT(8, LOCAL(17))),
    LABEL(CONSTANT(8, LOCAL(9))),
    LABEL(CONSTANT(8, LOCAL(13))),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    LABEL(CONSTANT(8, LOCAL(4))),
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
    LABEL(CONSTANT(8, LOCAL(11))),
    LABEL(CONSTANT(8, LOCAL(15))),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    LABEL(CONSTANT(8, LOCAL(10))),
    LABEL(CONSTANT(8, LOCAL(14))),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
  ),
)
RODATA(LOCAL(17), 8,
  LOCAL(9),
  LOCAL(11),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
)
