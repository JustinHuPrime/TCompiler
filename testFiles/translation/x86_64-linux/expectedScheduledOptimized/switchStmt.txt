testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    LABEL(LOCAL(3)),
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    J1B(LOCAL(3), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    J1A(LOCAL(3), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
    JUMPTABLE(TEMP(temp22, 8, 8, GP), LOCAL(17)),
    LABEL(LOCAL(9)),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
    LABEL(LOCAL(11)),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    LABEL(LOCAL(10)),
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
  ),
)
RODATA(LOCAL(17), 8,
  LABEL(LOCAL(9)),
  LABEL(LOCAL(11)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
  LABEL(LOCAL(10)),
  LABEL(LOCAL(9)),
)
