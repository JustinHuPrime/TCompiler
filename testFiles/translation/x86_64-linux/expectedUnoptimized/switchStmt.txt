testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    JUMP(CONSTANT(8, LOCAL(3))),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    JUMP(CONSTANT(8, LOCAL(7))),
  ),
  BLOCK(9,
    JUMP(CONSTANT(8, LOCAL(13))),
  ),
  BLOCK(13,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(10,
    JUMP(CONSTANT(8, LOCAL(14))),
  ),
  BLOCK(14,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(11,
    JUMP(CONSTANT(8, LOCAL(15))),
  ),
  BLOCK(15,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(12,
    JUMP(CONSTANT(8, LOCAL(16))),
  ),
  BLOCK(16,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(0))),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(7,
    J2B(CONSTANT(8, LOCAL(3)), CONSTANT(8, LOCAL(18)), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
  ),
  BLOCK(18,
    J2A(CONSTANT(8, LOCAL(3)), CONSTANT(8, LOCAL(19)), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
  ),
  BLOCK(19,
    SUB(TEMP(temp20, 1, 1, GP), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    ZX(TEMP(temp23, 8, 8, GP), TEMP(temp20, 1, 1, GP)),
    UMUL(TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP), CONSTANT(8, LONG(8))),
    ADD(TEMP(temp22, 8, 8, GP), TEMP(temp21, 8, 8, GP), CONSTANT(8, LOCAL(17))),
    JUMPTABLE(TEMP(temp22, 8, 8, GP), CONSTANT(8, LOCAL(17))),
  ),
  BLOCK(4,
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
  ),
)
RODATA(LOCAL(17), 8,
  LOCAL(9),
  LOCAL(11),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
  LOCAL(10),
  LOCAL(9),
)
