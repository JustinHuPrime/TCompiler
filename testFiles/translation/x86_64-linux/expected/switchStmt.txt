testFiles/translation/x86_64-linux/input/switchStmt.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    JUMP(LABEL(.L3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    JUMP(LABEL(.L7)),
  ),
  BLOCK(9,
    JUMP(LABEL(.L13)),
  ),
  BLOCK(13,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(31))),
    JUMP(LABEL(.L4)),
  ),
  BLOCK(10,
    JUMP(LABEL(.L14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(30))),
    JUMP(LABEL(.L4)),
  ),
  BLOCK(11,
    JUMP(LABEL(.L15)),
  ),
  BLOCK(15,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(28))),
    JUMP(LABEL(.L4)),
  ),
  BLOCK(12,
    JUMP(LABEL(.L16)),
  ),
  BLOCK(16,
    MOVE(TEMP(temp2, 1, 1, GP), CONSTANT(1, BYTE(0))),
    JUMP(LABEL(.L4)),
  ),
  BLOCK(7,
    JB(LABEL(.L3), LABEL(.L18), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
  ),
  BLOCK(18,
    JA(LABEL(.L3), LABEL(.L19), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(12))),
  ),
  BLOCK(19,
    SUB(TEMP(temp20, 1, 1, GP), TEMP(temp8, 1, 1, GP), CONSTANT(1, BYTE(1))),
    ZX(TEMP(temp23, 8, 8, GP), TEMP(temp20, 1, 1, GP)),
    UMUL(TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP), CONSTANT(8, LONG(8))),
    ADD(TEMP(temp22, 8, 8, GP), TEMP(temp21, 8, 8, GP), LABEL(.L17)),
    JUMP(TEMP(temp22, 8, 8, GP)),
  ),
  BLOCK(4,
    MOVE(REG(rax, 1), TEMP(temp2, 1, 1, GP)),
    RETURN(),
  ),
)
RODATA(.L17, 8,
  LABEL(.L9),
  LABEL(.L11),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
  LABEL(.L10),
  LABEL(.L9),
)
